examples/ros/src/smarts_ros/scripts/ros_driver.py

```diff
@@ -502,7 +502,10 @@ def _do_agents(self, observations: Dict[str, Observation]) -> Dict[str, Any]:
             for agent_id, agent in self._agents_to_add.items():
                 spec, mission = agent[0], agent[1]
                 assert agent_id not in self._agents
-                self._agents[agent_id] = spec.build_agent()
+                agent = spec.build_agent()
+                # XXX: hack! in the future this injection must be removed or done another way...
+                agent.sim = self._smarts
```

Comment:
Honestly, looking at it again I would still hope to not put this here.

Reply:
I agree! 
Let's try to think of a better way in the near future.